paper/P2728.md

```diff
@@ -999,22 +997,21 @@ You can also substitute a different replacement character by changing the
 result of the `else` clause, or add exception-based error handling by throwing
 at that point.
 
-## Caching `begin()`
-
-When we transcode from UTF-8, invoking `begin()` on the transcoding view may read up to
-four elements from the underlying view. Similarly, when transcoding from UTF-16, it may
-read up to two underlying elements. As a result, in order to preserve "amortized O(1)
-complexity," we need to cache `begin()` in these situations.
+## Why We Don't Cache `begin()`
 
-On the other hand, when transcoding from UTF-32, `begin()` reads at most one element from
-the underlying view, so `begin()` is not cached when doing so.
+When we invoke `begin()`, constructing the transcoding iterator may read up to four
+elements from the underlying view if it's transcoding from UTF-8. A previous revision of
+this paper implemented `begin()` caching, based on the idea that iterating the underlying
+range could have unbounded compexity.
 
-We also incorporate [@P3725R1]'s approach, in that when a transcoding view wraps a
-const-iterable `input_range` that is not a `forward_range`, the transcoding view provides
-a `const` overload of `begin()` that is non-caching.
+However, Tim Song pointed to the wording in `[iterator.requirements.general]` stating that
+"All the categories of iterators require only those functions that are realizable for a
+given category in constant time (amortized)." This means that we should be making the
+assumption that the underlying iterator operations used by `begin()` are "amortized
+constant time" (in a hand-wavey sense). Tim also pointed out that transcoding from UTF is
+equivalent to `views::adjacent<4>`, which doesn't cache.
```

Comment:
This is very interesting.  Does this mean that to_utfN will again be borrowable?